### PR TITLE
Add checkbox to PR template for What's new entry

### PR DESCRIPTION
**Pull request by hamogu**
See #9676 for discussion

### Description

This pull request is to address ...

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a What's new entry worthwhile and if so, is a "what's new" entry included in this PR?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by hamogu** [comment]
Aarg. Now it fails the bot that checks if a milestone it set, but since this is not code, it's just GH hiddne files that contol the PR template, I think it should not have a milestone.
Too many bots!

**Comment by pllim** [comment]
The milestone bot only appears when it wants to... 🤷

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (pllim) [inline .github/PULL_REQUEST_TEMPLATE.md:66]: Should we also recommend maintainer add the `whatsnew-needed` label here?

Comment (hamogu) [inline .github/PULL_REQUEST_TEMPLATE.md:66]:
```suggestion
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
```

**Review by pllim — APPROVED** [review]
LGTM but not sure if you want to keep it open for other reviews. Thanks!

Maybe squash?

Also you can apply `[ci skip]` to the commit. I don't think we need to run the CI for this.

**Comment by hamogu** [comment]
I'll leave this open for at least a day or two in case other people want to comment. I'll squish it after that.

**Comment by pllim** [comment]
It's been 5 days. Is it time to squish and merge?

**Comment by hamogu** [comment]
I already squished / squashed / squashed. Feel free to merge when you feel like it.

**Comment by pllim** [comment]
Awesome, thanks!

**Pull request merged**
